consensus/polybft/state_sync_manager.go

```diff
@@ -319,24 +326,22 @@ type PostEpochRequest struct {
 // so that it can discard any previous epoch commitments, and build a new one (since validator set changed)
 func (s *stateSyncManager) PostEpoch(req *PostEpochRequest) error {
 	s.lock.Lock()
-	defer s.lock.Unlock()
+
+	s.pendingCommitments = nil
+	s.validatorSet = req.ValidatorSet
+	s.epoch = req.NewEpochID
 
 	// build a new commitment at the end of the epoch
 	nextCommittedIndex, err := req.SystemState.GetNextCommittedIndex()
 	if err != nil {
 		return err
```

Comment:
is it intensional to keep holding lock in case of that return?

Reply:
That's a mistake from my side. Good catch :muscle:

Reply:
Fix: https://github.com/0xPolygon/polygon-edge/pull/1083/commits/a26ea1dc1a24ce4b625357b5d53237f5a149808e